R/functions-xcmsSwath.R

```diff
@@ -0,0 +1,33 @@
+## function "align" EICs
+.alignEics <- function(x, y) {
+
+  # linear interpolation of y on x
+  y_aligned <- approx(rtime(y), intensity(y), rtime(x))
+  
+  # deal with NA
+  y_aligned[is.na()] <- 0
```

Comment:
I would rather keep the `NA`s . Otherwise a subsequent correlation would correlate an intensity with a 0, if it was `NA` (because there is no signal for that specific retention time) the value pair could be skipped.

Reply:
True. So far I always replaced them with zeros. Can we agree to have an option like `na.replace` and if this is `TRUE` than we have `0` otherwise `NA`. Default value will be `FALSE`

Reply:
Sure. And what about an option `naValue = NA` and replace `NA` with that value (only `if (!is.na(naValue)) y_aligned[is.na(y_aligned)] <- naValue`)?